pkg/receive/writer.go

```diff
@@ -105,13 +115,18 @@ func (r *Writer) Write(ctx context.Context, tenantID string, wreq *prompb.WriteR
 		if ref == 0 {
 			// If not, copy labels, as TSDB will hold those strings long term. Given no
 			// copy unmarshal we don't want to keep memory for whole protobuf, only for labels.
-			labelpb.ReAllocZLabelsStrings(&t.Labels, r.intern)
+			labelpb.ReAllocZLabelsStrings(&t.Labels, r.opts.Intern)
 			lset = labelpb.ZLabelsToPromLabels(t.Labels)
 		}
 
 		// Append as many valid samples as possible, but keep track of the errors.
 		for _, s := range t.Samples {
-			ref, err = app.Append(ref, lset, s.Timestamp, s.Value)
+			if tooFarInFuture != 0 && tooFarInFuture.Before(model.Time(s.Timestamp)) {
+				// now + tooFarInFutureTimeWindow < sample timestamp
+				err = storage.ErrOutOfBounds
```

Comment:
add debug log for how long it out of bounds?

Reply:
Could we even make it Warning level? That is something that should fail a bit louder i think so errors can be investigated better.